### PR TITLE
Scroll header toolbar to end when RTL

### LINKED ISSUE #3889: RTL: "+" icon not in view when paragraph block is focused
**Describe the bug**
When I have tapped into a paragraph block, I expect to see the "+" icon, visible in order to add a new block. Instead, the "+" icon is scrolled off to the right so I cannot see it.
This happens when using the app in an RTL language, on Android only.

**To Reproduce**
1. Go to Me > App Settings > Interface Language and choose an RTL language such as Arabic or Hebrew.
2. Navigate back to Posts and open a post that has some blocks in it.
3. Note that the "+" icon appears at the bottom-right when you have selected an image block, embed block, or other block with few items in the edit bar (where undo/redo and other items appear).
3. Add an empty paragraph or other text-based block, and tap into it.
4. Note that the "+" icon is not visible on the edit bar. In order to find it, scroll over to the right past the Bold, Indent, and Undo/Redo options.

**Expected behavior**
I expected the "+" icon to appear, with other options scrolled out of visibility. This is the case with LTR languages and for RTL languages on iOS.

**Screenshots**
Compare iOS (left) to Android (right):
IMG_FCB7B6C64137-1

**Smartphone:**
Samsung Galaxy S21, Android 11, WPAndroid 18.1-rc-1

**Additional context**
Discovered while beta-testing WP Android 18.1

### PR DESCRIPTION
**Pull request by hypest**
Fixes #3889

### Related PRs
* https://github.com/WordPress/gutenberg/pull/34617

### To test

1. Set the device/emulator to an RTL language like Hebrew
2. Run the demo app on a device narrow enough to not be able to show the main toolbar when a text block is selected
3. Tap to select a text block(for instance, paragraph or header) in the demo content
4. Notice that the main toolbar scrolls to reveal the inserter button. See the video in the GB PR as an example.

### PR submission checklist

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

**Review by SiobhyB — APPROVED** [review]
Approved via https://github.com/WordPress/gutenberg/pull/34617#pullrequestreview-748224750

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Unreleased
 ---
 * [**] Enable WordPress embed preview [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3853]
+* [**] Fix main toolbar initial position when RTL [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3923]
```
Comment (SiobhyB) [inline RELEASE-NOTES.txt]:
```suggestion
* [**] [Android only] Fix main toolbar initial position when RTL [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3923]
```

Similar to my comment on the Gutenberg PR, do you think adding an `Android only` indicator here is a good idea?

**Review by SiobhyB — APPROVED** [review]
Approved via https://github.com/WordPress/gutenberg/pull/34617#pullrequestreview-751263975, with a similar small suggestion/question for the RELEASE-NOTES, but no blockers.

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (hypest) [reply RELEASE-NOTES.txt]: Yeap, rephrased in https://github.com/wordpress-mobile/gutenberg-mobile/pull/3923/commits/bc31dfae5853158035bbf1a1522f50499f3b1dc2, let me know what you think!

Comment (SiobhyB) [reply RELEASE-NOTES.txt]: Looks good to me! 🚢 🚀

**Pull request merged**